recipe/build.sh

```diff
@@ -1,26 +1,49 @@
 #!/bin/bash
+set -euxo pipefail
+
+mkdir build-${HOST} && pushd build-${HOST}
 
 if [[ $target_platform =~ linux.* ]]; then
-    USE_ICONV=--without-iconv
+    USE_ICONV=-DENABLE_ICONV=FALSE
 else
-    USE_ICONV=--with-iconv
+    USE_ICONV=-DENABLE_ICONV=TRUE
 fi
 
-autoreconf -vfi
-mkdir build-${HOST} && pushd build-${HOST}
-${SRC_DIR}/configure --prefix=${PREFIX}  \
-                     --with-zlib         \
-                     --with-bz2lib       \
-                     ${USE_ICONV}        \
-                     --with-lz4          \
-                     --with-lzma         \
-                     --with-lzo2         \
-                     --with-zstd         \
-                     --without-cng       \
-                     --with-openssl      \
-                     --without-nettle    \
-                     --with-xml2         \
-                     --without-expat
-make -j${CPU_COUNT} ${VERBOSE_AT}
+cmake .. ${CMAKE_ARGS}                                 \
+    -DCMAKE_PREFIX_PATH=$PREFIX                        \
+    -DCMAKE_INSTALL_PREFIX=$PREFIX                     \
+    -DCMAKE_BUILD_TYPE=Release                         \
+    -DCMAKE_C_FLAGS_RELEASE="$CFLAGS"                  \
+    -DENABLE_TEST=FALSE                                \
+    -DENABLE_ZLIB=TRUE                                 \
+    -DENABLE_BZIP2=TRUE                                \
+    -DBZIP2_ROOT=$PREFIX/lib                           \
+    -DENABLE_LZ4=TRUE                                  \
+    -DENABLE_LZMA=TRUE                                 \
+    -DENABLE_LZO=TRUE                                  \
+    -DENABLE_ZSTD=TRUE                                 \
+    -DENABLE_CNG=FALSE                                 \
+    -DENABLE_NETTLE=FALSE                              \
+    -DENABLE_XML2=TRUE                                 \
+    -DENABLE_EXPAT=FALSE                               \
+    $USE_ICONV
+
+make -j${CPU_COUNT}
 make install
-popd
+
+
+if [[ $target_platform =~ linux.* ]]; then
+    # backwards compatibility with artifacts <= version=3.5.2,build=2
+    MAJOR_VERSION=$(echo $PKG_VERSION | cut -d. -f1)
+    MINOR_VERSION=$(echo $PKG_VERSION | cut -d. -f2)
+    MINOR_PATCH_VERSION=$(echo $PKG_VERSION | cut -d. -f2-)
+    # Check upstream CMakeLists.txt for details
+    if [[ $MAJOR_VERSION == 3 ]]; then
+        SONAME_BASE=13
+    else
+        echo "MAJOR_VERSION $MAJOR_VERSION not recognized. Update build.sh to specify its SONAME_BASE"
```

Comment:
I'm assuming we won't need any repo patching or rebuilds, right? This ensure the old packages built with this will work.

I'm just concern of the sustainability of this `else` but I guess this is better than causing more churn in the ecosystem.

Reply:
Existing packages should work on Linux, but there's no workaround on macOS :(

See [this comment](https://github.com/conda-forge/libarchive-feedstock/issues/69#issuecomment-1152256077)

Reply:
Also I'd say we should only add the symlinks in this transition PR. Next release (e.g. 3.6) should not add the buggy sonames and just stick to upstream conventions (should be so.19 by then).

Reply:
I'm OK with that. Let's wait for some else to comment before merging just so we have third opinion.

@jakirkham you are the only other maintainer here BTW :grimacing:

Reply:
In my opinion the autoconf workflow has a bug. The expected so names are single integers according to their own docs:

* [Configure.ac](https://github.com/libarchive/libarchive/blob/464f9d9f1a5b3f51845e5f8662a37f48a7dfd996/configure.ac#L35-L47)
* [CMakeLists.txt](https://github.com/libarchive/libarchive/blob/464f9d9f1a5b3f51845e5f8662a37f48a7dfd996/CMakeLists.txt#L71-L77)

It's unclear to me why we ended up with these weird `13.5.2` so names. Looks like `13` + the minor.patch version, string-concatenated, instead of arithmetically summed. CMake does produce `18`.

There have been some similar issues in other distros, in which they couldn't agree on the soname version, but all of them use single integers:

* https://github.com/AppImage/AppImageKit/issues/343, https://github.com/AppImage/AppImageKit/issues/476
* https://github.com/libarchive/libarchive/issues/817